web/server/codechecker_server/server.py

```diff
@@ -912,6 +915,46 @@ def add_product(self, orm_product, init_db=False):
 
         self.__products[prod.endpoint] = prod
 
+    def get_if_database_in_use(self, product):
+        """
+        Returns the product endpoint if the given database is in use
+        """
+        # get the database name from the product connection string
+        is_sqlite = False
+        if product.database.endswith('.sqlite'):
+            to_add = product.database.replace('////', '/')
+            is_sqlite = True
+        else:
+            to_add = f"@{product.host}:{product.port}/{product.database}"
+
+        LOG.info("The database string that will be added: %s", to_add)
+
+        LOG.info("to add: %s", to_add)
+
+        # get the current state of connected databases from products
+        dynamic_list = [
+            a.connection.replace('////', '/')
```

Comment:
SQLAlchemy can be used for parsing the connection string: https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.engine.make_url
This could be used for a better check condition in the loop at line 951.

Reply:
fixed